### PR TITLE
Set order of tenant_template to the highest to avoid being overridden by custom templates

### PR DESCRIPTION
**Pull request by cliu123**
Signed-off-by: Chang Liu <lc12251109@gmail.com>

### Description
The issue description of https://github.com/opensearch-project/security-dashboards-plugin/issues/1246 explains the issue to be resolved.
The PR title described the approach to fix the issue. 

### Category
Bug fix

### Issues Resolved
https://github.com/opensearch-project/security-dashboards-plugin/issues/1246

### Testing
UTs and ITs

### Check List
- [x] New functionality includes testing
- [x] Commits are signed per the DCO using --signoff

### Screenshots
Screen Shot 2022-12-08 at 5 09 43 PM

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by DarshitChanpura — CHANGES_REQUESTED** [review]
@cliu123 Thank you for adding this PR. Two quick things:
1. Can you create a global constant for the order? (too bad javascript doesn't have a built-in for Java's Integer.MAX_VALUE)
2. Can you add a test showing that tenant template is not overridden in both scenarios where order is equal and is less than 2147483647?

File: server/multitenancy/tenant_index.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ export async function setupIndexTemplate(
     await esClient.indices.putTemplate({
       name: 'tenant_template',
       body: {
+        order: 2147483647,
```
Comment (peternied) [inline server/multitenancy/tenant_index.ts:44]: Can you add a short inline comment as to why this value should be high?

File: server/multitenancy/test/tenant_index.test.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+describe('Tenant index template', () => {
```
Comment (peternied) [inline server/multitenancy/test/tenant_index.test.ts:16]: Thanks for adding a test, is there any way to test the broader scenario has been fixed, e.g. since the order is high when a new index is created it gets X value instead of Y value?

File: server/multitenancy/tenant_index.ts (hunk 1 above)
Comment (peternied) [reply server/multitenancy/tenant_index.ts:44]: Note; If you create a const for the order value as DC recommended with a meaningful name that would also be enough to protect the 'why' of this value.

File: server/multitenancy/test/tenant_index.test.ts (hunk 2 above)
Comment (cliu123) [reply server/multitenancy/test/tenant_index.test.ts:16]: The template and order are parts of OpenSearch core. security-dashboards-plugin only calls the API to create template. It’s sufficient to make sure that the highest order is being set.

**Comment by cliu123** [comment]
> 1. Can you create a global constant for the order? (too bad javascript doesn't have a built-in for Java's Integer.MAX_VALUE)

👍  Done

> 2. Can you add a test showing that `tenant_template` is not overridden in both scenarios where order is equal and is less than 2147483647?

As I replied to the other comment, the template and order are parts of OpenSearch core. security-dashboards-plugin only calls the API to create template. So the definition of template and order, and how it gets overridden are defined by OpenSearch core. There's no need to test OpenSearch core functionality in plugins(would be duplicate effort).

**Review by peternied — APPROVED** [review]
Thanks @cliu123!

**Review by RyanL1997 — APPROVED** [review]

**Comment by davidlago** [comment]
Thanks for the work, @cliu123 !

@DarshitChanpura just double-checking that you had a chance to re-review after the latest change as the last commit addressed them, even though you didn't formally re-approve. Thanks!
